posthog/models/message_preferences.py

```diff
@@ -0,0 +1,146 @@
+from django.db import models
+from django.core.validators import EmailValidator
+from django.core.signing import TimestampSigner, SignatureExpired, BadSignature
+from typing import Optional
+
+
+class MessageCategory(models.Model):
+    key = models.CharField(max_length=64, unique=True)
+    name = models.CharField(max_length=128)
+    description = models.TextField(blank=True)
+    is_system_category = models.BooleanField(default=False, help_text="System categories cannot be deleted")
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        verbose_name_plural = "message categories"
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class RecipientIdentifier(models.Model):
+    IDENTIFIER_TYPE_CHOICES = [
+        ("email", "Email Address"),
+        ("phone", "Phone Number"),
+        ("device", "Device ID"),
+        ("push", "Push Token"),
+        ("browser", "Browser Token"),
+        ("slack", "Slack User ID"),
+    ]
+
+    identifier = models.CharField(max_length=512)
+    type = models.CharField(max_length=32, choices=IDENTIFIER_TYPE_CHOICES)
```

Comment:
I actually don't like this choices thing as it ends up being a DB migration every time we modify it. WE can just do this validation at the API layer